src/coreclr/pal/src/thread/context.cpp

```diff
@@ -618,7 +618,7 @@ void CONTEXTToNativeContext(CONST CONTEXT *lpContext, native_context_t *native)
             FPREG_Xmm(native, i) = lpContext->FltSave.XmmRegisters[i];
         }
 #elif defined(HOST_ARM64)
-#ifdef TARGET_OSX
+#if defined(TARGET_OSX) || defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
```

Comment:
❓ should we introduce a common define for this to simplify the conditions, e.g. `TARGET_APPLE`

Reply:
Yes, please (`*APPLE` or `*OSXLIKE`).

Reply:
+1

Reply:
There's a precedent in `OperatingSystem.IsOSXLike()`, `PlatformDetection.IsOSXLike` internal APIs, `IsOSXLike` used as property in CoreLib MSBuild, and NativeAOT using `IsOSXLike` in several places. Mono used `TARGET_APPLE_MOBILE` as a C #define but that's not exact match and it's not used anywhere else.

Ultimately either one is fine and understandable.

Reply:
I'd rather not introduce new defines with OSX in the name :-)

We have `TARGET_DARWIN` as a CMake variable in mono but I think I prefer `TARGET_APPLE` as it's clearer when you're not as familiar with the history of Apple platforms. Will add the define.